pyhub/mcptools/microsoft/excel/com/excel.py

```
@@ -0,0 +1,782 @@
+            self._app.Visible = visible
+            self._app.DisplayAlerts = False  # Suppress alerts
+
+            if add_book and self._app.Workbooks.Count == 0:
+                self._app.Workbooks.Add()
+
+        except Exception as e:
+            # Clean up COM if initialization failed
+            if self._com_initialized:
+                pythoncom.CoUninitialize()
+                self._com_initialized = False
+            raise ExcelError(f"Failed to initialize Excel application: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)